packages/itwinui-react/vitest.config.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    globals: true,
+    environment: 'jsdom',
+    setupFiles: 'setupTests.ts',
+    coverage: {
+      provider: 'v8',
+    },
+    exclude: ['/node_modules/', '/esm/', '/cjs/'],
+    reporters: ['junit', 'default'],
+    outputFile: 'coverage/junit.xml',
+    sequence: {
+      hooks: 'list',
+    },
+  },
+  resolve: {
+    alias: [
+      {
+        find: /^(.*)\/styles.js$/,
+        replacement: `${__dirname}/src/__mocks__/stylesMock.cjs`,
+      },
+      {
+        find: /^(.*)\/useGlobals.js$/,
+        replacement: `${__dirname}/src/__mocks__/useGlobalsMock.cjs`,
+      },
+    ],
+  },
```

Comment:
This is quite a roundabout way of adding a mock.

A simpler approach would be to directly use `vi.mock` inside `setupTests.ts`, then remove the `__mocks__` folder.

For example:
```ts
vi.mock('./src/styles.js', () => {
  return {
    default: new Proxy(
      {},
      {
        get: (target, prop) => {
          // instead of returning scoped css modules, we will preserve the original classes
          if (typeof prop === 'string' && prop.startsWith('iui')) {
            return prop;
          }
          return Reflect.get(target, prop);
        },
      },
    ),
  };
});
```

Reply:
The [CI](https://github.com/iTwin/iTwinUI/actions/runs/7670132012/job/20905879601?pr=1784) is failing:

<img width="987" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/1e9fde33-c209-41dd-a0b9-8434dd05a84e">

Reply:
The path for `useGlobals` is wrong

```diff
- vi.mock('./src/useGlobals.js', () => {
+ vi.mock('./src/core/utils/hooks/useGlobals.js', () => {
```

Reply:
Yeah in that case, I'd say bring back what you had with aliases, except use [`test.alias`](https://vitest.dev/config/#alias) instead of `resolve`

Also, I think it's only needed for `styles.js` (meaning `useGlobals` might be fine as just `vi.mock`).

Lastly, the `__mocks__` folder would make more sense outside of `src` and let's use the name `styles.js` (instead of `stylesMock.cjs`) so that it's ESM and matches the name of the thing we are mocking.

Putting it altogether:
```js
// vitest.config.ts
import { defineConfig } from 'vitest/config';
import * as path from 'node:path';

export default defineConfig({
  test: {
    // ...
    alias: [
      {
        find: /^(.*)\/styles.js$/,
        replacement: path.resolve('./__mocks__/styles.js'),
      },
    ],
  },
  // ...
});
```

```js
// __mocks__/styles.js
export default new Proxy(
  {},
  {
    get: (target, prop) => {
      // instead of returning scoped css modules, we will preserve the original classes
      if (typeof prop === 'string' && prop.startsWith('iui')) {
        return prop;
      }
      return Reflect.get(target, prop);
    },
  },
);
```

Reply:
The `css: false` flag might not be working. Try adding this to the `mock-css` vite plugin:

```js
load(id) {
  if (id.endsWith('styles.css')) {
    return '';
  }
  return null;
},
```